compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -2893,14 +2843,27 @@ private boolean checkFillerValue(BUnionType type) {
         if (type.isNullable()) {
             return true;
         }
-        Iterator<BType> iterator = type.getMemberTypes().iterator();
+        Set<BType> typeTags = new HashSet<>();
```

Comment:
Type tags mean something else for the compiler right? Shall we use a different name?